AutomatedTesting/Gem/PythonTests/assetpipeline/fbx_tests/pythonassetbuildertests.py

```diff
@@ -0,0 +1,94 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+"""
+
+# Import builtin libraries
+import pytest
+import logging
+import os
+import stat
+
+# Import LyTestTools
+from ly_test_tools.o3de.asset_processor import AssetProcessor
+from ly_test_tools.o3de import asset_processor as asset_processor_utils
+import ly_test_tools.environment.file_system as fs
+
+# Import fixtures
+from ..ap_fixtures.asset_processor_fixture import asset_processor as asset_processor
+from ..ap_fixtures.ap_setup_fixture import ap_setup_fixture as ap_setup_fixture
+
+# Import LyShared
+from ly_test_tools.o3de.ap_log_parser import APLogParser, APOutputParser
+import ly_test_tools.o3de.pipeline_utils as utils
+
+# Use the following logging pattern to hook all test logging together:
+logger = logging.getLogger(__name__)
+# Configuring the logging is done in ly_test_tools at the following location:
+# ~/dev/Tools/LyTestTools/ly_test_tools/log/py_logging_util.py
+
+# Helper: variables we will use for parameter values in the test:
+targetProjects = ["AutomatedTesting"]
+
+@pytest.fixture
+def local_resources(request, workspace, ap_setup_fixture):
+    ap_setup_fixture["tests_dir"] = os.path.dirname(os.path.realpath(__file__))
+
+
+@pytest.mark.usefixtures("asset_processor")
+@pytest.mark.usefixtures("ap_setup_fixture")
+@pytest.mark.usefixtures("local_resources")
+@pytest.mark.parametrize("project", targetProjects)
+@pytest.mark.assetpipeline
+@pytest.mark.SUITE_main
+class TestsPythonAssetProcessing_APBatch(object):   
+
+    @pytest.mark.BAT
+    @pytest.mark.assetpipeline
+    def test_ProcessAssetWithoutScriptAfterAssetWithScript_ScriptOnlyRunsOnExpectedAsset(self, workspace, ap_setup_fixture, asset_processor):
+        # This is a regression test. The situation it's testing is, the Python script to run
+        # defined in scene manifest files was persisting in a single builder. So if
+        # that builder processed file a.fbx, then b.fbx, and a.fbx has a Python script to run,
+        # it was also running that Python script on b.fbx.
+
+        asset_processor.prepare_test_environment(ap_setup_fixture["tests_dir"], "TwoSceneFiles_OneWithPythonOneWithout_PythonOnlyRunsOnFirstScene")
+
+        asset_processor_extra_params = [
+        # Disabling Atom assets disables most products, using the debugOutput flag ensures one product is output.
+        "--debugOutput",
+        # By default, if job priorities are equal, jobs run in an arbitrary order. This makes sure
+        # jobs are run by sorting on the database source name, so they run in the same order each time
+        # when this test is run.
+        "--sortJobsByDBSourceName",
```

Comment:
I am wondering if this should be done by default, or will this affect source dependency?

Reply:
It won't effect source dependencies, it only triggers when everything else is equal. The old logic defaults to using the job ID to sort, which is basically arbitrary at that point.

I'm hesitant to make it the default because that's a pretty big change in functionality. I'll directly reach out to some other people to see what they think.